Examples/Scripts/FreeSurferPipelineBatch-long.sh

```diff
@@ -1,22 +1,22 @@
 #!/bin/bash
-
 function identify_timepoints
 {
     local subject=$1
     local tplist=""
     local tp visit n
 
-    #build the list of timepoints
+    #build the list of timepoints (sessions)
     n=0
-    for visit in ${PossibleVisits[@]}; do
-        tp="${subject}_${visit}"
-        if [ -d "$tp" ] && ! [[ " ${ExcludeVisits[*]} " =~ [[:space:]]${ExcludeVisits[*]}[[:space:]] ]]; then
-            if (( n==0 )); then 
-                tplist="$tp"
-            else
-                tplist="$tplist@tp"
-            fi
+    for session in ${PossibleVisits[*]}; do
+        tp="${subject}_${session}"
+        if [ -d "$tp" ] && ! [[ " ${ExcludeVisits[*]} " =~ [[:space:]]"$tp"[[:space:]] ]]; then
+             if (( n==0 )); then 
```

Comment:
`@` was the more-correct specifier in the `for` line (it works to produce separate tokens when inside double quotes).  Also, if the first visit is excluded, your logic will put an `@` on the start of the resulting string, because you increment `n` even when the visit is excluded.  It may be cleaner code to build an array, and then use the `IFS` trick to go back to an `@`-separated string afterwards (and that makes it easy to check whether all visits for a subject were excluded, warn, and skip to the next subject).  If you want to stick with the string way, the simpler solution is to test `if [[ "$tplist" == "" ]]`.

Also, a bare `((expression))` is NOT a good idea under `set -e` (or with debug.shlib), because if the result is `0`, it will act like a `false`, which the shell sees as an abnormal exit code, and will abort the script.  `if` specifically catches the exit status of whatever comes after it, and uses that to decide whether to execute the `then` or `else` section, which is why `if ((somevar))` works without aborting.